include/genomics/printer.hpp

```diff
@@ -162,66 +195,87 @@ namespace genomics {
     csvline += ",";
     csvline += strand;
 
-    csvline += ",";
-    csvline += complement(m.sequence);
-
     csvline += ",";
     csvline += std::to_string(m.mismatches);
 
-    csvline += ",";
-    csvline += std::to_string(m.rna_bulges);
+    if (complete) {
+      csvline += ",";
+      csvline += complement(m.sequence);
 
-    csvline += ",";
-    csvline += std::to_string(m.dna_bulges);
+      csvline += ",";
+      csvline += std::to_string(m.rna_bulges);
+
+      csvline += ",";
+      csvline += std::to_string(m.dna_bulges);
+    }
 
     return csvline;
   }
 
   template <class t_wt, uint32_t t_dens, uint32_t t_inv_dens>
   std::string get_csv_lines(const genome_index<t_wt, t_dens, t_inv_dens>& gi,
                             const kmer& k, bool start,
-                            std::vector<std::vector<std::tuple<int64_t, match>>>& off_targets) {
+                            int64_t max_off_targets,
+                            std::vector<std::vector<std::tuple<int64_t, match>>>& off_targets, bool complete) {
     std::string csvlines;
 
+    float cfd_sum = 0.0;
+    std::vector<std::string> off_targets_lines;
     for (uint64_t d = 0; d < off_targets.size(); d++) {
-      for (const auto& off_target : off_targets[d]) {
+      for (int64_t i = 0; i < off_targets[d].size(); i++) {
+        if ((max_off_targets != -1) && (i >= max_off_targets)) break;
+        const auto& off_target = off_targets[d][i];
         int64_t off_target_abs_coords = std::get<0>(off_target);
         match match = std::get<1>(off_target);
-        csvlines += get_csv_line(gi, k, start, match, off_target_abs_coords);
-        csvlines += "\n";
+        std::string csvline = get_csv_line(gi, k, start, match, off_target_abs_coords, complete);
```

Comment:
Slightly awkward because we have to wait till we get all off-target lines before calculating and appending cfd value to each.